tools/clang/test/CodeGenSPIRV/unary-op.sizeof.hlsl

```diff
@@ -0,0 +1,13 @@
+// Run: %dxc -T ps_6_0 -E main
+
```

Comment:
This is the PR that added `sizeof` support in the DXIL backend: https://github.com/microsoft/DirectXShaderCompiler/pull/2178/files

It contains a test named [`sizeof.hlsl`](https://github.com/microsoft/DirectXShaderCompiler/pull/2178/files#diff-6b02cc32328e2832e594e1d179253ef7). I suggest we add all the types from that test to the SPIR-V backend test too.

A test should try its best to target one thing. For example, if a bug related to ByteAddressBuffer is introduced later in the code, it will break the `sizeof` test because it's using it.

So, it's better to have the test only use `sizeof` on its own.

for example:

```cpp
// Run: %dxc -E main -T vs_6_2 -HV 2018 -enable-16bit-types

struct complexStruct { .... };

void main() {
  uint size = 0;
  // CHECK: ...
  size = sizeof(float3);
  // CHECK: ...
  size = sizeof(int16_t3[2]);
  // CHECK: ...
  size = sizeof(complexStruct);
}
```

Reply:
done